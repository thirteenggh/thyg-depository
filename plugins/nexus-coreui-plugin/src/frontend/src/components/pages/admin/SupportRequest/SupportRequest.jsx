import React from 'react';
import {ContentBody, NxButton, NxFontAwesomeIcon, Page, PageHeader, PageTitle, Section} from 'nexus-ui-plugin';
import {faExternalLinkAlt, faUserCircle} from '@fortawesome/free-solid-svg-icons';

import UIStrings from '../../../../constants/UIStrings';
import './SupportRequest.scss';

export default function SupportRequest() {
  function openSupportRequestPage() {
    window.open('https://links.sonatype.com/products/nexus/pro/support-request', '_blank');
  }

  return <Page>
    <PageHeader><PageTitle icon={faUserCircle} {...UIStrings.SUPPORT_REQUEST.MENU}/></PageHeader>
    <ContentBody className='nxrm-support-request'>
      <Section>
        <p>{UIStrings.SUPPORT_REQUEST.DESCRIPTION}</p>
        <p><a href="#admin/support/supportzip">{UIStrings.SUPPORT_REQUEST.ATTACH_SUPPORT_ZIP}</a></p>
        <NxButton variant="primary" onClick={() => openSupportRequestPage()}>
          <NxFontAwesomeIcon icon={faExternalLinkAlt}/>
          <span>{UIStrings.SUPPORT_REQUEST.ACTIONS.submitRequest}</span>
        </NxButton>
      </Section>
    </ContentBody>
  </Page>;
}
