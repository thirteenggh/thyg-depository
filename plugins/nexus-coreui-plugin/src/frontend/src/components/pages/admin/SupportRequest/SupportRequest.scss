
.nxrm-support-request {
  h2 {
    margin-top: 0
  }

  table {
    table-layout: fixed;
    width: 100%;
  }
}
