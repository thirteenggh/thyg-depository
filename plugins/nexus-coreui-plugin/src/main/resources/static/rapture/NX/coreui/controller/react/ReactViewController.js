/*global Ext, NX*/

/**
 * Anonymous Security Settings controller.
 *
 * @since 3.21
 */
Ext.define('NX.coreui.controller.react.ReactViewController', {
  extend: 'NX.app.Controller',

  views: [
    'react.MainContainer'
  ],

  refs: [
    {
      ref: 'reactMainContainer',
      selector: 'nx-coreui-react-main-container'
    },
    {
      ref: 'breadcrumb',
      selector: 'nx-breadcrumb'
    }
  ],

  listen: {
    controller: {
      '#Refresh': {
        refresh: 'refresh'
      }
    },
    component: {
      'nx-coreui-react-main-container': {
        render: function() {
          this.getBreadcrumb().hide();
        },
        destroy: function() {
          this.getBreadcrumb().show();
        }
      }
    }
  },

  refresh: function() {
    if (this.getReactMainContainer()) {
      this.getReactMainContainer().refresh();
    }
  }

});
