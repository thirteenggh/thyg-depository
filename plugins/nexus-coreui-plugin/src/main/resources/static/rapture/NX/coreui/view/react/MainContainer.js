/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  reactViewProps: null,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView, this.reactViewProps, null);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
