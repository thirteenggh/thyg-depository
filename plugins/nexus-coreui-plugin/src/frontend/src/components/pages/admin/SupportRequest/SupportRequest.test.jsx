import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SupportRequest from './SupportRequest';

describe('SystemInformation', () => {
  it('renders correctly', async () => {
    const {container} = render(<SupportRequest/>);

    expect(container).toMatchSnapshot();
  });
});
