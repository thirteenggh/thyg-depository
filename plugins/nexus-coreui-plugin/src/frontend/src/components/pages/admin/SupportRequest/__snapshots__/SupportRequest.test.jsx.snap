// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemInformation renders correctly 1`] = `
<div>
  <main
    class="nx-page-main"
  >
    <div
      class="nxrm-page-header"
    >
      <div
        class="nx-page-title"
      >
        <h1
          class="nx-h1 nx-feature-name"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user-circle fa-w-16 nx-page-title__page-icon nx-icon"
            data-icon="user-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
              fill="currentColor"
            />
          </svg>
          <span>
            Support Request
          </span>
        </h1>
        <p
          class="nx-page-title__description nx-feature-description"
        >
          Submit a support request to TianheCloud
        </p>
      </div>
    </div>
    <div
      class="nxrm-content-body nxrm-support-request"
    >
      <div
        class="nxrm-section nx-tile"
      >
        <p>
          Please include a complete description of your problem and steps to allow us to reproduce the problem (if available).
        </p>
        <p>
          <a
            href="#admin/support/supportzip"
          >
            Attaching a support ZIP to your request will help our engineers give you a faster response.
          </a>
        </p>
        <button
          class="nx-btn nx-btn--primary"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-external-link-alt fa-w-18 nx-icon"
            data-icon="external-link-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
              fill="currentColor"
            />
          </svg>
          <span>
            Submit Request
          </span>
        </button>
      </div>
    </div>
  </main>
</div>
`;
